scripts/ai/AIDriveStrategyPlowCourse.lua

```diff
@@ -45,55 +43,35 @@ function AIDriveStrategyPlowCourse.new(customMt)
     return self
 end
 
-function AIDriveStrategyPlowCourse:setAIVehicle(vehicle, jobParameters)
-    -- need to set the plow before calling the parent's setAIVehicle so if it calls any
-    -- overwritten functions, we have the plow set up already
-    self.plow = AIUtil.getAllChildVehiclesWithSpecialization(vehicle, Plow)[1]
-    AIDriveStrategyPlowCourse:superClass().setAIVehicle(self, vehicle, jobParameters)
-    if self:hasRotatablePlow() then
-        self:debug('has rotatable plow.')
-    end
-end
-
 function AIDriveStrategyPlowCourse:getDriveData(dt, vX, vY, vZ)
     if self.state == self.states.INITIAL then
         -- When starting work with a plow it first may need to be unfolded and then turned so it is facing to
         -- the unworked side, and then can we start working
-
         self:setMaxSpeed(0)
-        self:setOffsetX()
-
-        -- this will unfold the plow when necessary
-        self.vehicle:raiseAIEvent("onAIStart", "onAIImplementStart")
-        self.vehicle:requestActionEventUpdate()
-
-        if self.plow.getIsUnfolded and self.plow:getIsUnfolded() then
-            self:debug('Plow already unfolded, now rotating if needed')
-            self:rotatePlow()
-            self.state = self.states.ROTATING_PLOW
-        else
-            self:debug('Unfolding plow')
-            self.state = self.states.UNFOLDING_PLOW
-        end
+        self:updatePlowOffset()
+        if self:isPlowTurning() then
+            self:rotatePlows()

```

Comment:
I'm confused with this one, do we check if it is rotating (with isPlowTurning()) and if yes, then rotate it?